### PR TITLE
refactor(Core/Misc): Convert some enums to enum class

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
- Dont know if this is the best way, but here we are :D

## Tests Performed:

- builds

## How to Test the Changes:

1. Play around with stuff the enum's touch ingame

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
@Kitzunu build failing

**Comment by Kitzunu** [comment]
Is obly module. I will fix that later

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -631,11 +631,11 @@ void ObjectMgr::LoadCreatureTemplateResistances()
         Field* fields = result->Fetch();
 
         uint32 creatureID = fields[0].GetUInt32();
-        uint8 school = fields[1].GetUInt8();
+        SpellSchool school = SpellSchool(fields[1].GetUInt8());
 
-        if (school == SPELL_SCHOOL_NORMAL || school >= MAX_SPELL_SCHOOL)
+        if (school == SpellSchool::Normal || school >= SpellSchool::MaxSpellSchool)
```
Comment (Kitzunu) [inline src/server/game/Globals/ObjectMgr.cpp:636]:
```suggestion
        if (school == SpellSchool::Normal || school >= SpellSchool(MAX_SPELL_SCHOOL))
```

File: src/server/game/Entities/Unit/Unit.h [hunk 2]
```diff
@@ -17,6 +17,8 @@
 #include "ThreatManager.h"
 #include <functional>
 
+enum class DispelType : uint32;
+
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.h:21]:
```suggestion
```

File: src/server/shared/SharedDefines.h [hunk 3]
```diff
@@ -183,29 +183,30 @@ enum Powers
...
-    SPELL_SCHOOL_SHADOW                 = 5,
-    SPELL_SCHOOL_ARCANE                 = 6
+    Normal                 = 0,
+    Holy                   = 1,
+    Fire                   = 2,
+    Nature                 = 3,
+    Frost                  = 4,
+    Shadow                 = 5,
+    Arcane                 = 6,
+    MaxSpellSchool         = 7
```
Comment (Kitzunu) [inline src/server/shared/SharedDefines.h:195]:
```suggestion
```

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by Winfidonarleyan** [comment]
@Kitzunu conflcts

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by Kitzunu** [comment]
todo

**Comment by Kitzunu** [comment]
https://github.com/azerothcore/mod-duel-reset/pull/23

I have no idea why Eluna is failing. Nor am I surprised as eluna sucks.

**Comment by Kitzunu** [comment]
Well, only eluna left. And I have no idea how to fix it.

**Comment by FrancescoBorzi** [comment]
`mod-duel-reset` failing also with 

```
modules/mod-duel-reset/src/DuelReset_scripts.cpp:81:21: fatal error: use of undeclared identifier 'DUEL_WON'
        if (type == DUEL_WON) {
```

**Comment by Kitzunu** [comment]
image

**Comment by Kitzunu** [comment]
wtb help with eluna errors that I don't understand.

**Comment by Nyeriah** [comment]
My guess is that while Eluna's ```Push()``` would easily accept enums as they get morphed into anything, it has a hard time dealing with enum class, as they have a defined form. E.g ```Push(DUEL_WON)``` is fine but ```Push(DuelType)``` doesn't pass

**Comment by Kitzunu** [comment]
better take them one by one isntead of bulk action

**Pull request closed without merging**